tests/functional/test_summary.py

```diff
@@ -0,0 +1,87 @@
+#
+from time import sleep
+
+import pytest
+from datetime import timedelta
+
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet
+from test_utils.size import Size, Unit
+from utils.iotrace import IotracePlugin
+
+mount_point = "/mnt/cas"
+test_file_path = f"{mount_point}/test_file"
+iterations_per_config = 10
+cache_size = Size(16, Unit.GibiByte)
+
+runtime = timedelta(seconds=8)
+label = "super_iotracer_test_instance"
+
+
+@pytest.mark.require_disk("device", DiskTypeSet(
+    [DiskType.sata, DiskType.hdd4k, DiskType.hdd]))
+def test_compare_summary():
+    """
+        title: Compare trace summaries.
+        description: |
+          Verify the summary returned at the end of tracing
+          (output of 'iotrace –S') is the same as the summary read
+          from trace files (output of 'iotrace -M -G').
+        pass_criteria:
+          - Trace summaries are the same.
+    """
+    with TestRun.step("Prepare io-tracer."):
+        iotrace: IotracePlugin = TestRun.plugins["iotrace"]
+        disk = TestRun.disks['device']
+
+    with TestRun.step("Run tracing for a while"):
+        output = iotrace.parse_json(
+            TestRun.executor.run(
+                f'iotrace -S -d {disk.system_path} -t 8').stdout)[-1]
+
+    with TestRun.step("Compare tracing summaries."):
+        summary = IotracePlugin.get_trace_summary(output["tracePath"])
+        if output != summary:
+            TestRun.LOGGER.error(
+                "The summary read from trace file differs from summary "
+                "returned at the end of tracing.")
+
+
+def test_check_summary():
+    """
+        title: Verify trace summary from file.
+        description: |
+          Verify the summary read from trace files (output of 'iotrace -M -G').
+        pass_criteria:
+          - Trace summary is correct.
+    """
+    with TestRun.step("Prepare io-tracer."):
+        iotrace: IotracePlugin = TestRun.plugins["iotrace"]
+
+    with TestRun.step("Run tracing for a while"):
+        iotrace.start_tracing(timeout=runtime, label=label)
+        sleep(int(runtime.total_seconds()) + 1)
+
+    with TestRun.step("Verify tracing summary."):
+        latest_path = IotracePlugin.get_latest_trace_path()
+        summary = IotracePlugin.get_trace_summary(latest_path)
+        fault = False
+        if summary["tracePath"] != latest_path:
+            fault = True
+        if summary["state"] != "COMPLETE":
+            fault = True
+        if summary["sourceNode"]["node"][0]["id"] \
+                != latest_path.split("/")[0]:
+            fault = True
+        if summary["traceStartDateTime"] \
+                != latest_path[-19:-3].replace("_", " "):
+            fault = True
+        if summary["traceDuration"] != str(int(runtime.total_seconds())):
+            fault = True
+        if summary["label"] != label:
+            fault = True
+        if fault:
+            TestRun.fail("Trace summary is invalid")
```

Comment:
Maybe change from having a single generic fail, but have an error/fail in all paths? In case of failure you don't know which (and/or how many) case caused it.

Reply:
Or concatenate to single error message in each "if" - so we test all "ifs" even when it fails on the first one.